src/bika/lims/browser/auditlog.py

```diff
@@ -190,14 +195,28 @@ def folderitems(self):
         # slice a batch
         batch = snapshots[self.limit_from:self.limit_from+self.pagesize]
 
-        for snapshot in batch:
+        for num, snapshot in enumerate(batch):
             item = self.make_empty_item(**snapshot)
+
             # get the version of the snapshot
-            version = get_snapshot_version(self.context, snapshot)
+            #
+            # NOTE: We need to calculate the version based on the known index,
+            # because at this point we have similar snapshots that would return
+            # the same version!
+            #
+            # With this PR we added a timestamp to the metadata of the snapshot
+            # to make sure they can be distinguished:
+            # https://github.com/senaite/senaite.core/pull/2739
+            #
+            # version = get_snapshot_version(self.context, snapshot)
+            version = self.total - self.limit_from - num - 1
 
             # Version
             item["version"] = version
 
+            snapshot_data = json.dumps(snapshot, indent=2, sort_keys=True)
+            item["snapshot"] = api.text_to_html(snapshot_data, wrap="pre")
```

Comment:
:clap:  Really cool to have the full snapshot in JSON format directly accessible in the listing!